Formula/stencil.rb

```diff
@@ -0,0 +1,19 @@
+class Stencil < Formula
+  desc "Configurable static site generator"
+  homepage "https://getoutreach.github.io/stencil/"
+  url "https://github.com/getoutreach/stencil/archive/refs/tags/v1.12.1.tar.gz"
+  sha256 "bf8f342963362d580cf75afb8608b1bffe93ad0b320df7d9b21ef365d8332c6e"
+  license "Apache-2.0"
+  head "https://github.com/getoutreach/stencil.git", branch: "main"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-s -w -X github.com/getoutreach/gobox/pkg/app.Version=v1.12.1"),
+      "./cmd/stencil"
+  end
+
+  test do
+    system "#{bin}/stencil", "--help"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.